docs/core/diagnostics/eventsource.md

```diff
@@ -20,6 +20,17 @@ your own custom events.
 > Many technologies that integrate with EventSource use the terms 'Tracing' and 'Traces' instead of 'Logging' and 'Logs'.
 > The meaning is the same here.
 
+> [!WARNING]
+> Potential Deadlocks with EventSource Callbacks
```

Comment:
In addition to deadlocks folks probably also need to beware of infinite recursion and using types before they are fully initialized.

Reply:
Updated